packages/core/http/core-http-server/src/versioning/types.ts

```diff
@@ -9,6 +9,7 @@
 
 import type { ApiVersion } from '@kbn/core-http-common';
 import type { MaybePromise } from '@kbn/utility-types';
+import { VersionedRouterRoute } from '@kbn/core-http-router-server-internal';
```

Comment:
⚠️ circular dependency!
`VersionRouterRoute` [depends on types](https://github.com/elastic/kibana/blob/d70583faddd3b5f3f3e8d59888a8bebdb262a4d2/packages/core/http/core-http-router-server-internal/src/versioned_router/types.ts#L10-L15) from `core-http-server`.
I'd move `VersionedRouterRoute` to `core-http-common`.

Reply:
For example:

```//  core-http-server/src/versioning/types.ts
/**
 * Public counterpart to Internal type
 */
export interface PublicVersionedRouterRoute {
  method: string;
  path: string;
  options: Omit<VersionedRouteConfig<RouteMethod>, 'path'>;
  handlers: Array<{ fn: RequestHandler; options: AddVersionOpts<unknown, unknown, unknown> }>;
  versionsOptions: Record<string, AddVersionOpts<unknown, unknown, unknown>>;
  isVersioned: true;
}
```

then in the internal package:

``` // core-http-server-internal/src/versioned_router/types.ts
import type {
  AddVersionOpts,
  RequestHandler,
  RouteMethod,
  PublicVersionedRouterRoute,
} from '@kbn/core-http-server';
...

/** @internal */
export type VersionedRouterRoute = Omit<
  PublicVersionedRouterRoute,
  'handlers' | 'versionsOptions'
> & {
  handlers: Array<{ fn: RequestHandler; options: Options }>;
  versionsOptions: Record<string, Options>;
};
```